### ISSUE TITLE
feature request: Allow markdown files to be processed in templates

### ISSUE THREAD

**Issue by wolftune** [yesodweb/shakespeare#199]
Simplest version is to just allow markdown files to be used where they get run through pandoc or similar (ideally a setting for whoever to choose the processor) and then the resulting HTML is used where the processed Hamlet file would otherwise be used.

Possible alternate or addition could be a Hamlet command that specifies that the block within that command is markdown and should be processed as such.

Ideal: recognize hamlet functions like typesafe URLs and such within a markdown file so that the combination of markdown and type safety is possible.

The primary goal here is to make it even more approachable for Yesod projects to reach wider audience of contributors who may be more comfortable with markdown than HTML. But there's other potential uses and advantages, especially the readability factor of the code as readable text.

**Comment by snoyberg** [comment]
It's a nice idea in theory. It's unclear how we'd handle control
structures, and also whether this should go in Shakespeare or a different
package due to dependency issues.

But fortunately we already have a name for it: Markus.

**Comment by erikd** [comment]
If you're using Roman names shouldn't it be "Marcus" instead of "Markus"?

**Comment by snoyberg** [comment]
Yes, but the bad spelling is good for disambiguation and ties into markdown
more obviously. But I'm open to the other working.

**Comment by chreekat** [comment]
By the way, I haven't looked into it yet, but http://github.com/arowM/heterocephalus seems to be moving in a direction I think could be valuable.

**Comment by cdepillabout** [comment]
@chreekat If you have any requests/suggestions for heterocephalus feel free to open an issue/PR.

(Pinging @arowM as well.)
